zhaquirks/tuya/ts0043.py

```diff
@@ -192,6 +192,99 @@ class TuyaSmartRemote0043TO(CustomDevice):
     }
 
 
+class TuyaSmartRemote0043TOPlusA(CustomDevice):
+    """Tuya 3-button remote device with time on out cluster."""
+
+    signature = {
+        MODEL: "TS0043",
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    OnOff.cluster_id,
+                    TuyaZBE000Cluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
+            },
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    PowerConfiguration.cluster_id,
+                    OnOff.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
+            3: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    PowerConfiguration.cluster_id,
+                    OnOff.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
+            4: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+                INPUT_CLUSTERS: [
+                    PowerConfiguration.cluster_id,
+                    OnOff.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
```

Comment:
So the device only has three switches but four endpoints? (Hence you removed the 4th endpoint in the replacement?)

The same signature is used by a four button ``TS0044`` device:
https://github.com/zigpy/zha-device-handlers/blob/961567488e9456c59691340814af817776b85dce/zhaquirks/tuya/ts0044.py#L232-L281
So I guess Tuya just lazily renamed the model whilst keeping the fourth endpoint.

I think this should be good to go though, as it's a different model name (so we shouldn't be worried that a four button switch is matched accidentally).